tests/test_data_catalog.py

```diff
@@ -294,6 +293,16 @@ def test_from_archive(tmpdir):
         data_catalog.from_archive("https://asdf.com/asdf.zip")
 
 
+def test_from_yml_with_archive(tmpdir):
+    yml_fn = join(CATALOGDIR, "artifact_data.yml")
+    data_catalog = DataCatalog(yml_fn)
+    sources = list(data_catalog.sources.keys())
+    assert len(sources) > 0
+    root = dirname(data_catalog[sources[0]].path)
+    yml_dst_fn = join(root, "artifact_data.yml")
+    assert isfile(yml_dst_fn)
```

Comment:
it's not super clear what this test actually asserts to me, shouldn't we also assert that the catalog is actually cached to disk?